dataRetriever/interface.go

```diff
@@ -49,7 +49,7 @@ const (
 // Resolver defines what a data resolver should do
 type Resolver interface {
 	RequestDataFromHash(hash []byte) error
-	ProcessReceivedMessage(message p2p.MessageP2P, broadcastHandler func(buffToSend []byte)) error
+	ProcessReceivedMessage(message p2p.MessageP2P, fromConnectedPeer p2p.PeerID, broadcastHandler func(buffToSend []byte)) error
```

Comment:
"from" should be enough.

Reply:
It can be easily mistaken with message.From . fromConnectedPeer explains better.